lib/Text/Regex/TDFA/NewDFA/Engine.hs

```diff
@@ -706,7 +707,10 @@ updateCopy ((_i1,instructions),oldPos,newOrbit) preTag s2 i2 = do
 
 -- #ifdef __GLASGOW_HASKELL__
 foreign import ccall unsafe "memcpy"
-    memcpy :: MutableByteArray# RealWorld -> MutableByteArray# RealWorld -> Int# -> IO (Ptr a)
+    memcpyIO :: MutableByteArray# RealWorld -> MutableByteArray# RealWorld -> Int# -> IO (Ptr a)
+
+memcpyST :: MutableByteArray# s -> MutableByteArray# s -> Int# -> State# s -> (# State# s, Ptr a #)
+memcpyST = unsafeCoerce memcpyIO
```

Comment:
Is the suffix `ST` connected to the `ST` monad?  
If yes, good, if no, please rename.
Because when I see `fooIO` and `fooST` side-by-side I think of two versions of `foo`, one for the `IO` and the other for the `ST` monad.

Reply:
Yes:

```haskell
newtype ST s a = ST (STRep s a)
type STRep s a = State# s -> (# State# s, a #)
```

Reply:
Excellent!